src/track/beatiterator.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "track/beats.h"
+#include "track/beatutils.h"
+
+namespace mixxx {
+/// A forward iterator for Beats.
+/// BeatIterator do not fulfills the Iterator semantics, use with care.
+class BeatIterator final {
+  public:
+    BeatIterator(BeatList::const_iterator start, BeatList::const_iterator end)
+            : m_currentBeat(start),
+              m_endBeat(end),
+              m_ended(false) {
+        // Advance to the first enabled beat.
+        while (m_currentBeat <= m_endBeat && !m_currentBeat->enabled()) {
+            ++m_currentBeat;
+        }
+    }
+
+    ~BeatIterator() = default;
+
+    using iterator_category = std::forward_iterator_tag;
+
+    bool hasNext() const {
+        return !m_ended;
+    }
+
+    /// Advances de iterator and returns the current beat frame position.
+    /// If you need the frame position make sure you store it, is not possible
+    /// to get it again.
+    double next() {
+        double beat = m_currentBeat->frame_position();
+        if (beat != m_endBeat->frame_position()) {
+            ++m_currentBeat;
+            while (beat != m_endBeat->frame_position() && !m_currentBeat->enabled()) {
+                ++m_currentBeat;
+            }
+        } else {
+            m_ended = true;
+        }
+        return beat;
+    }
+
+    // TODO(hacksdump): These will be removed from this iterator and
+    // will be made methods of the Beat class to be introduced in #2844.
+
+    /// Returns true if the current beat is a down beat.
+    bool isBar() const {
+        return m_currentBeat->type() == mixxx::track::io::BAR;
+    }
+
+    /// Returns true if the current beat is a phrase beat.
+    bool isPhrase() const {
+        return m_currentBeat->type() == mixxx::track::io::PHRASE;
+    }
+
+    /// The current beat becomes a down beat.
```

Comment:
The comments if makeBeat() and makeBar() don't seem to be reasonable. 

Aren't beats that are marked as mixxx::track::io::BAR supposed to be a downbeat, i.e. first beat in a bar/measure?